src/integration/README.md

```diff
@@ -14,38 +14,90 @@ This will build and run the application that the tests will run against.
+```js
+export default () => {
+  describe('Analytics', () => {
+    it('ATI', () => {
+      const noscriptImage = document.querySelector('noscript img');
+
+      expect(noscriptImage.tagName).toEqual('IMG');
+      expect(noscriptImage.getAttribute('width')).toEqual('1px');
+      expect(noscriptImage.getAttribute('height')).toEqual('1px');
+      expect(noscriptImage.getAttribute('src')).toMatch(
+        'https://logws1363.ati-host.net?',
+      );
+    });
+  });
+};
+```
+
+For Jest to run the tests in these files we need to create a `*(amp|canonical).test.js` file that will import and call the test functions. We use `amp.test.js` to run the tests on the AMP platform and `canonical.test.js` to run the tests on the canonical platform. For example a file name `amp.test.js` with the contents:
 
 ```js
 /**
- * @pathname /mundo/articles/ce42wzqr2mko
+ * @service pidgin
+ * @pathname /pidgin/23248703
  */
 
-it('I can see the headline', () => {
-  const headlineEl = canonical.getByText(headlineText);
+import runCrossPlatformTests from '../crossPlatformTests';
+import runAmpTests from '../ampTests';
 
-  expect(headlineEl).toBeInTheDocument();
+describe(platform.toUpperCase(), () => {
+  describe(pageType, () => {
+    runCrossPlatformTests();
+    runAmpTests();
+  });
 });
 ```
 
+In the above example we import the cross platform tests and the AMP tests. We have also specified a pathname using a [docblock pragma](#what-is-a-docblock-pragma). The pathname is the part of the url that is everything after `https://bbc.com` and in this example it is `/mundo/articles/ce42wzqr2mko`. If you visit `https://bbc.com/mundo/articles/ce42wzqr2mko` (**NB** this is the canonical url - for the AMP url just add `.amp` on the end) you will see this is a Mundo article page and it is what we are going to test.
+
+Before our tests run, the test environment [setup file](https://github.com/bbc/simorgh/tree/latest/src/integration/integrationTestEnvironment.js) parses the `pathname` dockblock pragma and constructs the url. JSDOM then visits the url to get the DOM trees that we can use to run our tests against.
+
+Note we have also specified a `service` docblock pragma. The service is parsed and added to the global scope of every test file. This can come in handy for tests where you need to know the service. There are other useful variables added to the global scope:
+
+- `pageType` - The page type which is parsed from the test file path.
+- `service` - The service of the currently running page test.
+- `document` - The `document` object of the current page. You can use this to query the DOM and assert the correct things are in the page.
+- `platform` - The current platform (`amp` or `canonical`)
```

Comment:
Is there any need to have this in the global scope? Isn't that implicit knowledge based on whether you are writing and AMP or canonical test? And I guess a cross platform test shouldn't matter. 

I'm just wary of cross-platform tests containing `if(platform === 'amp')` rather than tests being correctly put in their respective platform related files.

Reply:
that's a fair point!